pylint/lint/base_options.py

```diff
@@ -103,7 +103,7 @@ def _make_linter_options(linter: PyLinter) -> Options:
                 "short": "f",
                 "group": "Reports",
                 "help": "Set the output format. Available formats are text,"
-                " parseable, colorized, json and msvs (visual studio)."
+                " parseable, colorized, json2 and msvs (visual studio)."
```

Comment:
I don't think we should hide the fact that the json reporter still exists, generating this text would be nice too imo.

Reply:
I guess we have a different opinion here then.

I think there should be one way to do things and that `json2` is superior. Thus, that option is what we should advertise to users looking through the possibilities.

Reply:
Yes, but imagine you maintain a tool that consumes the old reporter. And you're reviewing a PR, and you need to brush up on the context. You come here, and it's gone. Won't you burn time trying to figure out what happened and wondering why things aren't horribly broken?

Reply:
Isn't that what the changelog is for? If you search for `json` you'll find the change.

_I'm fine with adding something like `json (deprecated)` but as expressed elsewhere I would really like to nudge all users to the new reporter._

Reply:
Yeah I'm waffling on this one. As long as it's somewhere I guess it doesn't have to be in the CLI help.